### DISCUSSION TITLE
Tagging

### DISCUSSION THREAD

**Discussion by simonv3** [funmusicplace/mirlo#166] [Ideas]
Right now we have no tagging, but we should figure out tagging! 

I feel like at the top level albums should be tag-able. But should artists as well? 

And should there be different types of tags? Location and other? Or can those be lumped together?

**Comment by LouisLouiseKay** [comment]
By the look of it Bandcamp only has keywords and location tags. https://bandcamp.com/tags

Their search system is quite comprehensive though
![Screenshot_2023-10-25-21-43-38-125_com bandcamp android-edit](https://github.com/funmusicplace/mirlo/assets/77789607/c7baac6d-5284-448b-8823-8599489a1ccc)

Probably using a mix of tags and other Metadata (pop > synth pop > in France > in Vinyl format for example)

**Reply by simonv3** [reply]
I kind of dislike the bandcamp tag search system, but the tagging itself is probably fine.

**Reply by Cwpute** [reply]
why do you dislike ? i find it very usable as a listener. it favours discovery as well: i like pop music so i search for pop, but oh what is that ? bedroom pop ? let's have a listen.
i like that each "great currents" gets divided into its own subsidiaries

**Comment by meljoann** [comment]
As a listener, I'd say I don't mind if an artist doesn't have their own genre tags. You're just looking for new tunes.

As an artist, it's hard to commit to artist genre tags that could shift from release to release.

Don't think location needs to be separate. Some artists won't feel comfortable putting their location -- or it could be a complicated issue if you've emigrated etc. But location tag can be useful as a listener, if you're looking for what's going in in a particular scene.

**Reply by Cwpute** [reply]
i agree about location tags: sometimes you want to listen to music from a specific area

**Comment by samnabi** [comment]
- I'm in favour of location tags (by city ideally, but if artists want to be more general they can tag their country)
    - Location tags on Bandcamp are incredibly helpful for booking/discovering local bands or digging into a particular city's scene
- I like genre tags at an artist level, can we do some clever grouping behind the scenes so that similar tags (i.e. `hip-hop`, `Hip Hop`, `hiphop`) are all treated the same way?
    - If an artist wants to edit the genre tags on a particular album or song, they should be able to do that, but if there are none defined it falls back to the artist tags.

Picking up on the discussion from Discord, I think it would be really cool for listeners to define their own genre tags. I wouldn't want this to show up on the artist's bio or profile, but maybe it could just be used when searching/browsing.

**Reply by simonv3** [reply]
@samnabi would you be able to give me a screenshot of what location tagging looks on bandcamp for an artist?

**Reply by samnabi** [reply]
From what I can tell, Bandcamp suggests the City name as a pre-populated tag when you create an album. The downside with this is, it's treated the same as a genre tag, and doesn't distinguish between different cities of the same name.

<img width="986" alt="Screen Shot 2024-01-15 at 6 24 34 PM" src="https://github.com/funmusicplace/mirlo/assets/1731536/dc1944ad-fc7c-441a-87a0-557a06dbb9a1">

<img width="577" alt="Screen Shot 2024-01-15 at 6 27 28 PM" src="https://github.com/funmusicplace/mirlo/assets/1731536/41f1b21c-97e7-4355-aa57-f607633f157a">

Instead, if you want to find artists from a particular city, you have to use the location filter on the [discover](https://bandcamp.com/discover) page. It would be better if clicking on the city in the artist's bio took you to a listing of other artists in that city.

<img width="351" alt="Screen Shot 2024-01-15 at 6 28 07 PM" src="https://github.com/funmusicplace/mirlo/assets/1731536/803f521b-0ecc-4e05-9537-dfd168d475ae">

Bonus points: if there are no other artists in that city, show artists from that State/Province. If there are still no matching artists, show artists from that country.

**Reply by simonv3** [reply]
Thanks for the detailed insights! I feel like this is getting into slightly complicated territory, and might be worth building out as its own feature. I think there's a lot of value in _nearby_ artists finding each other, as well as finding artists near you as a listener, but at the moment it might be a little sparse? There's also questions to be asked here about how we handle a federated system of places. Like, what does a discovery feature look like that does locality and looks for artists across a couple of platforms (say jam.coop, faircamp, etc)/

**Reply by timglorioso** [reply]
Listener tags:

Thinking about artist/listener interaction. Maybe have them show up in their own category on artist and/or album pages? Clear distinction with some label like "Listeners tagged this with"

Give artists some way to manage or respond, possibly an opt-in feature

**Comment by simonv3** [comment]
Commenting this here:

https://mirlo.space/releases?tag=rock

(though I do notice there's a bug with the header displaying `<strong>`)

And you can see them on albums too

https://mirlo.space/avalanches/release/megalithe.

I'm not going to close this because there's a couple of other features suggested above that I think will be Good To Have

* [x] Tags that have a location meaning
* [x] Tagging artists so that album tags fall back on an artist's tag
* [x] Normalizing some tags, or at least tag search, so that similar tags return results

**Comment by Cwpute** [comment]
imo, i really like what bandcamp did with their tagging system and tag search recommendations. Ii really helps you discover stuff, and makes for fun dives into genres you don't know.

I believe the way they generate the second line of tags is by looking at the *other tags* on releases that feature the tag you searched for.
![image](https://github.com/funmusicplace/mirlo/assets/80604512/a087569b-aeb4-4f66-991d-f7e5a8acebc8)
For example in that previous example: you searched for releases tagged with `kitchener`, and are shown all releases tagged with it. the second line of tags, next to `all kitchener`is generated by looking at all releases with a `kitchener` tag and picking among the *other tags* these releases are tagged with.

A good way to generate that second line more thoughtfully would probably be to have the first recommended tag be the first or second most common used tag, and then all others be random tags.

**Comment by simonv3** [comment]
I'm closing this because tagging is implemented and fairly robust, and we just released location tags. We can discuss better searching of tags and locations elsewhere imo.
